src/QUESTIONS.md

```diff
@@ -0,0 +1,401 @@
+
+
+### Is my software safer just by running it on a CHERI-extended CPU?
+
+No. While an operating system (i.e., a kernel, basic programs and libraries
+shipped with the OS) might be compiled for the pure-capability ABI and hence
+prevent memory-safety bugs in its components, code compiled for a baseline
+architecture (e.g., AArch64 for Morello, RISC-V for CHERI-RISC-V) does not
+benefit from security features of CHERI.
+
+You must compile your code to explicitly use CHERI capabilities to access memory
+either by compiling it for the pure-capability ABI or the hybrid ABI with
+appropriate data type qualifiers that indicate the intention of using CHERI
+capabilities in place of legacy C pointers (in case of C/C++).
+
+
+### What are the pure-capability and hybrid ABIs?
+
+The pure-capability ABI (also known as CheriABI in CheriBSD and PCuABI in Linux
+and Android) is a new process ABI that uses only CHERI capabilities to access
+memory and interact with a kernel.
+
+The hybrid ABI allows a process to access memory using legacy pointers and CHERI
+capabilities.
+When compiling such C/C++ code, a pointer must be annotated with the
+`__capability` type qualifier to use a CHERI capability instead of a legacy
+pointer.
+
+Both pure-capability and hybrid CheriBSD kernels support the pure-capability and
+hybrid ABIs at the same time.
+You can also switch between pure-capability and hybrid CheriBSD kernels without
+replacing user-space programs and libraries.
+
+See the
+[CheriABI](https://www.cl.cam.ac.uk/research/security/ctsrd/pdfs/201904-asplos-cheriabi.pdf)
+paper and the
+[PCuABI](https://git.morello-project.org/morello/kernel/linux/-/wikis/Morello-pure-capability-kernel-user-Linux-ABI-specification)
+specification for more information on the ABIs.
+
+
+### Should I compile my code for the pure-capability or hybrid ABI?
+
+It might seem that compiling code for the hybrid ABI is a better approach to
+port software to CHERI than compiling it for the pure-capability ABI because you
+can incrementally introduce CHERI capabilities in your code, and not worry about
+all incompatibilities with CHERI at once.
+
+However, when using the hybrid ABI, you have to annotate all pointers as CHERI
+capabilities, and still you will not have any guarantees that there are no
+legacy pointers left.
+Additionally, all third-party libraries that your code depends on will not use
+CHERI capabilities, unless you change them as well.
+
+There are still some situations when choosing the hybrid ABI over the
+pure-capability ABI has benefits.
+In case of extremely large code bases (e.g., the Chromium browser and its
+dependencies), you might want to compartmentalize parts of your code base now
+(e.g., the V8 JavaScript engine in Chromium), parts that are known for security
+vulnerabilities with significant impacts, rather than spend a lot of time
+trying to port the whole project for the pure-capability ABI and then
+compartmentalize it.
```

Comment:
Do we believe this is a good example? I'm skeptical on the grounds that we've been fighting with V8 for many months.